Portage/PackageId.hs

```diff
@@ -109,64 +123,36 @@ toCabalPackageId (PackageId (PackageName _cat name) version) =
   fmap (Cabal.PackageIdentifier name)
            (Portage.toCabalVersion version)
 
-instance Text Category where
-  disp (Category c) = Disp.text c
-  parse = fmap Category (Parse.munch1 categoryChar)
-    where
-      categoryChar c = Char.isAlphaNum c || c == '-'
-
-instance Text PackageName where
-  disp (PackageName category name) =
-    disp category <> Disp.char '/' <> disp name
-
-  parse = do
-    category <- parse
-    _ <- Parse.char '/'
-    name <- parse
-    return (PackageName category name)
-
-instance Text PackageId where
-  disp (PackageId name version) =
-    disp name <> Disp.char '-' <> disp version
-
-  parse = do
-    name <- parse
-    _ <- Parse.char '-'
-    version <- parse
-    return (PackageId name version)
-
 parseFriendlyPackage :: String -> Maybe (Maybe Category, Cabal.PackageName, Maybe Portage.Version)
 parseFriendlyPackage str =
-  case [ p | (p,s) <- Parse.readP_to_S parser str
-       , all Char.isSpace s ] of
-    [] -> Nothing
-    (x:_) -> Just x
+  case explicitEitherParsec parser str of
```

Comment:
This parser is failing to parse correctly if a version is specified in the package string to be parsed.